File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.py
class Solution:
    
    def sortVowels(self, s: str) -> str:
        b=[False]*len(s)
        vow=[]
        for i in range(len(s)):
            if(s[i] in "aeiouAEIOU"):
                b[i]=True
                vow.append(s[i])
        vow.sort()
        res=""
        ind=0
        for i in range(len(s)):
            if(b[i]==0):
                res+=s[i]
            else:
                res+=vow[ind]
                ind+=1
        return res
        

File: 2887-sort-vowels-in-a-string/sort-vowels-in-a-string.java
class Solution {
    public String sortVowels(String s) {
        String vow = "";
        ArrayList<Character> al = new ArrayList<>();
        String vowels = "aeiouAEIOU";
        for(char ch : s.toCharArray()){
            if(vowels.indexOf(ch) != -1){
                al.add(ch);
            }
        }
        Collections.sort(al);
        int i = 0;
        String res = "";
        for(char ch : s.toCharArray()){
            if(vowels.indexOf(ch) != -1){
                res += al.get(i);
                i++;
            }
            else
                res += ch;
        }
        return res;
    }
}